components/cldr-json-data-provider/src/cldr_langid.rs

```diff
@@ -0,0 +1,60 @@
+use icu_locale::LanguageIdentifier;
+use serde::{Deserialize, Deserializer};
+
+// TODO: Make this non-pub
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct CldrLanguage(pub LanguageIdentifier);
+
+impl<'de> Deserialize<'de> for CldrLanguage {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct CldrLanguageVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for CldrLanguageVisitor {
+            type Value = CldrLanguage;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                write!(formatter, "a valid Unicode Language Identifier or 'root'")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                if s == "root" {
```

Comment:
Is `"root"` a magic string that we can declare as a constant with a meaningful name?

Perhaps the same could be true of `"und"` in the next line.

Reply:
`"root"` is a string defined by CLDR LanguageIdentifier, which is the file we are in.  I think it's okay to have it in the body of the code as long as it is in this particular file.